api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -788,6 +792,9 @@ public static <T> T visit(Type type, CustomOrderSchemaVisitor<T> visitor) {
             new VisitFuture<>(map.keyType(), visitor),
             new VisitFuture<>(map.valueType(), visitor));
 
+      case VARIANT:
+        return visitor.variant();
```

Comment:
I'm debating whether this should pass the type. If we want to do it later, that would be an added method. When returning types, it's a good idea to pass the original instance, rather than assuming that it is identical.